profiles/lib/tag_cache.lua

```diff
@@ -0,0 +1,35 @@
+TagCache = {}
+
+function TagCache.get(data, key)
+  local v = data.cache[key]
+  if v then
+    if v == '' then
+      return nil
+    else
+      return v
+    end
+  else
+    v = data.way:get_value_by_key(key)
+    if v == nil then
+      data.cache[key] = ''
```

Comment:
in case the tag value is nil, then we store an empty string, so that next time the tag is requested, we know that we already fetched it